actionpack/lib/action_dispatch/testing/assertions/response.rb

```diff
@@ -85,17 +86,13 @@ def normalize_argument_to_redirection(fragment)
           end
         end
 
-        def generate_response_message(type)
-          message = "Expected response to be a <#{type}>, but was"
-
+        def response_message
           if @response.redirection?
             redirect_is = normalize_argument_to_redirection(@response.location)
-            message << " a redirect to <#{redirect_is}>"
+            "<302> redirected to <#{redirect_is}>"
```

Comment:
It needs to use the actual `@response.response_code`... my point was that there are multiple statuses (a hundred of them, in fact) that are `redirection?`.

Come to think of it, a bunch of them don't even have a location header.